### PR TITLE
Integrity tests for combats.txt plus fix for failing test because of Barrels change.

### PR DESCRIPTION
**Pull request by jaadams5**
This makes https://github.com/kolmafia/kolmafia/pull/1157 OBE although the discussion of test environments being different might still have some relevance,

### CODE REVIEW

File: test/net/sourceforge/kolmafia/DataFileConsistencyTest.java [hunk 1]
```diff
@@ -120,4 +123,74 @@ public void testFamiliarSpecificEquipment() throws IOException {
...
+    String[] fields;
+    try (BufferedReader reader = FileUtilities.getVersionedReader(file, version)) {
+      while ((fields = FileUtilities.readData(reader)) != null) {
+        if (fields.length == 1) {
+          continue;
+        }
+        // Second field is a Combat frequency - -1 or 0 <= x <= 100
+        String freq = fields[1];
+        int iFreq = StringUtilities.parseInt(freq);
+        boolean allowed = (iFreq == -1) || ((0 <= iFreq) && (iFreq <= 100));
```
Comment (gausie) [inline test/net/sourceforge/kolmafia/DataFileConsistencyTest.java:159]: I would just assertThat(freq, greaterThanOrEqualTo(-1)) and assertThat(freq, lessThanOrEqualTo(100))

File: test/net/sourceforge/kolmafia/DataFileConsistencyTest.java [hunk 2]
```diff
@@ -120,4 +123,74 @@ public void testFamiliarSpecificEquipment() throws IOException {
...
+  public void testValidMonstersForCombats() {
+    String file = "combats.txt";
+    int version = 1;
+    String[] fields;
+    try (BufferedReader reader = FileUtilities.getVersionedReader(file, version)) {
+      while ((fields = FileUtilities.readData(reader)) != null) {
+        if (fields.length == 1) {
+          continue;
+        }
+        // Third to end are monsters or
```
Comment (gausie) [inline test/net/sourceforge/kolmafia/DataFileConsistencyTest.java:177]: Or what 😛

**Review by gausie — COMMENTED** [review]
Thanks! Just a few comments

File: test/net/sourceforge/kolmafia/DataFileConsistencyTest.java (hunk 2 above)
Comment (jaadams5) [reply test/net/sourceforge/kolmafia/DataFileConsistencyTest.java]: Whoops

File: test/net/sourceforge/kolmafia/DataFileConsistencyTest.java (hunk 1 above)
Comment (jaadams5) [reply test/net/sourceforge/kolmafia/DataFileConsistencyTest.java]: OK

**Review by gausie — APPROVED** [review]

**Pull request merged**